core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -328,13 +330,24 @@ trait EntityComponent {
 
       // Active actions: only return entities and attributes with their deleted flag set to false
 
+      private def listTypeSql(workspaceContext: Workspace, entityType: String) =
+        sql"""#${baseEntityAndAttributeSql(workspaceContext)}
+        where e.deleted = false
+        and e.entity_type = ${entityType}
+        and e.workspace_id = ${workspaceContext.workspaceIdAsUUID}"""
+
+      // almost the same as "activeActionForType" except 1) adds a sort by e.id; 2) returns a stream
+      def activeStreamForType(workspaceContext: Workspace,
+                              entityType: String
+      ): SqlStreamingAction[Seq[EntityAndAttributesResult], EntityAndAttributesResult, Read] =
+        concatSqlActions(listTypeSql(workspaceContext, entityType), sql" order by e.id").as[EntityAndAttributesResult]
+
+      // TODO: if only used by tests, can this be removed?
```

Comment:
I suggest doing this in a follow-on PR to keep PRs managable

Reply:
Could you just move them into the test files as utility functions for now? Or would the follow-up PR have more in mind with these functions?

Reply:
I could refactor to move these into test-only code - that would still involve a good number of lines of code changing, and I wanted to limit the number of lines changing in this PR. But if you think it's worth it I'll do it in this PR instead of in a follow-on!

Reply:
Next PR sounds good!